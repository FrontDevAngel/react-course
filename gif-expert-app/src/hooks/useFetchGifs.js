import { useState, useEffect } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
    // use... en useFetchHooks es un standard en el nombre de hooks
    const [state, setState] = useState({
        data: [],
        loading: true,
    });

    useEffect(() => {
        getGifs(category).then((imgs) => {
            setState({
                data: imgs,
                loading: false,
            });
        });
    }, [category]);

    return state; // regresa el state;
};
