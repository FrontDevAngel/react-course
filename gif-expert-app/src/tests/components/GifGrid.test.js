import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../Components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas a componente <GifGrid />', () => {
    const category = 'Fc Barcelona';

    test('Hacer match del componente con el SnapShot', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category={category} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan las imagenes useFetchGifs', () => {
        const gifs = [
            {
                id: '1122334455',
                url: 'https://localhost/prueba/2323.png',
                title: 'Prueba de hooks',
            },
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });

        const wrapper = shallow(<GifGrid category={category} />);

        // expect(wrapper).toMatchSnapshot();

        // Busca si el elemento existe
        expect(wrapper.find('p').exists()).toBe(false);

        // Busca si tenemos componentes existentes y su cantidad
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
