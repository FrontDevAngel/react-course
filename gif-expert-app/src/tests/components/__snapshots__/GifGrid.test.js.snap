// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a componente <GifGrid /> Hacer match del componente con el SnapShot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Fc Barcelona
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
